FHEM/00_SIGNALduino.pm

```diff
@@ -2939,16 +2950,57 @@ sub SIGNALduino_FW_saveWhitelist
 		$wl_attr =~ s/,$//;			# Komma am Ende entfernen
 	}
 	$attr{$name}{whitelist_IDs} = $wl_attr;
-	SIGNALduino_Log3 $hash, 3, "$name Protocolist save: $wl_attr";
+	SIGNALduino_Log3 $name, 3, "$name Protocolist save: $wl_attr";
 	SIGNALduino_IdList("x:$name", $wl_attr);
 }
 
-sub SIGNALduino_FW_deselectAll
+sub SIGNALduino_FW_selectAll
 {
-	#my $hash = shift;
-	#my $name = $hash->{NAME};
-
-	#SIGNALduino_Log3 $hash, 3, "$name: button deselectAll";
+	my $name = shift;
+	my $hash = $defs{$name};
+	my %BlacklistIDs;
+	my @IdList = ();
+	my $ret = "";
+	my $bflag = 0;
+	my $devFlag = 0;
+	
+	my $blacklist = AttrVal($name,"blacklist_IDs","");
+	if (length($blacklist) > 0) {							# Blacklist in Hash wandeln
+		SIGNALduino_Log3 $name, 5, "$name Protocolist selectAll: attr blacklistIds=$blacklist";
+		%BlacklistIDs = map { $_ => 1 } split(",", $blacklist);
+		#my $w = join ', ' => map "$_" => keys %BlacklistIDs;
+		#SIGNALduino_Log3 $name, 3, "$name IdList, Attr blacklist $w";
+		$bflag = 1;
+	}
+	
+	my $develop = SIGNALduino_getAttrDevelopment($name);
+	if ($develop eq "1" || substr($develop,0,1) eq "y") {		# Entwicklerversion
+		$devFlag = 1;
+	}
+	
+	my $id;
+	foreach $id (keys %ProtocolListSIGNALduino)
+	{
+		next if ($id >= 900);
```

Comment:
wieso soll hier alles >= 900 übersprungen werden. Wozu ist das gut?

Reply:
das hatten wir schon weiter oben.
Dies ist eine Vorbereitung für einen weiteren Patch mit dem die Version der ProtocolList in den Internals angezeigt wird.
```

	"999" =>  # 
	{
		versionProtocolList  => 'dev_17.12.18'
	}
```

Reply:
Daß die IDs nur Zahlen sein dürfen, hatten wir weiter oben schon.
Die Versionsbezeichnung im Protocolhash werden wir auch bei der Aktualisierug des Protocolhash brauchen.
Wenn es Dir besser gefällt können wir die Version auch bei einer bestehenden ID wie z.B der ID 0 zufügen.

Reply:
Wir können ja einen Eintrag reservieren, die keys werden ohnehhin alle als Strings angelegt. hier alles >= 900 zu überspringen wäre falsch.
Ich haben in der Vergangenheit auch einen Fehler gemacht und nicht das Feld ID für die ID Verwendet sondern den Key eines Eintrages. Das rächt sich langsam :(

Reply:
ist es Dir lieber, wenn ich die ID 999 für die Version nehme und dann
`next if ($id == 999);`

Reply:
Mir ist es lieber wenn wir es hier erst Mal draussen lassen.

Reply:
Mir ist nicht klar wie die bessere Lösung aussehen soll?
- Du bist der Meinung, daß eine Versioninfo in dem Protokollhash überhaupt nicht nötig ist
- Die Versioninfo mit in die ID 0 eintragen
- in den Protokollhash mit einer nicht nummerischen ID wie z.B. "version" eine Unsauberkeit reinbringen. Diese Unsauberkeit wird uns höchstwahrscheinlich irgendwann wieder einholen.
`sort {$a <=> $b} @IdList; `ist dann nicht mehr möglich

Reply:
Wenn wir es so in den Protokollhash eintragen,
```
	"999" =>  # 
	{
		versionProtocolList  => 'dev_17.12.18'
	}
```
wie soll es dann zu Problemen kommen?

Reply:
Ok, dann kommt die versioninfo mit in die ID 0 rein bis Du eine bessere Lösung gefunden hast.
Die versioninfo soll so wie auch die version der 00_signalduino in einem Internal gespeichert werden.

> Das Ausgrenzen von IDs>900 führt nur dazu, dass die UnitTests nicht laufen.

Es soll nur die 999 ausgegrenzt werden.
Anscheinend siehst Du die Gefahr, daß jemand ausversehen die 999 für UnitTests verwendet.